packages/celotool/src/lib/prometheus.ts

```diff
@@ -115,8 +118,10 @@ async function createPrometheusGcloudServiceAccountforAKS(serviceAccountName: st
   }
 }
 
-function getServiceAccountNameforAKS(kubeClusterName: string) {
+function getServiceAccountName(clusterConfig: AzureClusterConfig | AwsClusterConfig) {
+  // Checks if resourceGroup, an Azure specific property, is present in config
+  const prefix = ('resourceGroup' in clusterConfig) ? 'aks' : 'aws'
```

Comment:
What do you think about creating a generic cluster config and having the Azure/AWS ones inherit from the generic one? Like

```
interface ClusterConfig {
  clusterName: string
  cloudProviderName: string
}

export interface AwsClusterConfig extends ClusterConfig {
  clusterRegion: string
}

export interface AzureClusterConfig extends ClusterConfig {
  // stuff
}
```

and then set `cloudProviderName` to `aks` or `aws` as necessary, and then you can just have `clusterConfig: ClusterConfig` and use `clusterConfig.cloudProviderName` instead of prefix here